controller/script/simulate-proxy/main.go

```diff
@@ -349,80 +379,51 @@ func main() {
 		log.Fatal(err.Error())
 	}
 
+	deploys, err := k8s.NewReplicaSetStore(clientSet)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
 	err = pods.Run()
 	if err != nil {
 		log.Fatal(err.Error())
 	}
 
-	podList, err := pods.List()
+	err = deploys.Run()
 	if err != nil {
 		log.Fatal(err.Error())
 	}
 
-	allPods := make([]*v1.Pod, 0)
-	podsByNamespace := make(map[string][]*v1.Pod)
-	for _, pod := range podList {
-		if pod.Status.PodIP != "" && !strings.HasPrefix(pod.GetNamespace(), "kube-") && (*maxPods == 0 || len(allPods) < *maxPods) {
-			allPods = append(allPods, pod)
-			podsByNamespace[pod.GetNamespace()] = append(podsByNamespace[pod.GetNamespace()], pod)
-		}
+	podList, err := pods.List()
+	if err != nil {
+		log.Fatal(err.Error())
 	}
 
+	deployments := getDeployments(podList, deploys, maxPods)
+
 	stopCh := make(chan os.Signal)
 	signal.Notify(stopCh, os.Interrupt, os.Kill)
 
+	excludedDeployments := make(map[string]struct{}, 0)
+
 	for _, addr := range strings.Split(*metricsAddrs, ",") {
-		randomNamespace := getRandomNamespaceKey(podsByNamespace)
-		proxyPod := podList[rand.Intn(len(podList))]
-
-		go func(address string, namespace string, pod *v1.Pod) {
-
-			proxy := simulatedProxy{
-				sleep:     *sleep,
-				pod:       pod,
-				namespace: namespace,
-				proxyMetricCollectors: &proxyMetricCollectors{requestTotals: prom.NewCounterVec(
-					prom.CounterOpts{
-						Name: "request_total",
-						Help: "A counter of the number of requests the proxy has received",
-					}, labels),
-					responseTotals: prom.NewCounterVec(
-						prom.CounterOpts{
-							Name: "response_total",
-							Help: "A counter of the number of responses the proxy has received.",
-						}, labels),
-					requestDurationMs: prom.NewHistogramVec(
-						prom.HistogramOpts{
-							Name:    "request_duration_ms",
-							Help:    "A histogram of the duration of a response",
-							Buckets: latencyBucketBounds,
-						}, labels),
-					responseLatencyMs: prom.NewHistogramVec(
-						prom.HistogramOpts{
-							Name:    "response_latency_ms",
-							Help:    "A histogram of the total latency of a response.",
-							Buckets: latencyBucketBounds,
-						}, labels)},
-			}
+		randomPodOwner := getRandomDeployment(deployments, excludedDeployments)
+		excludedDeployments[randomPodOwner] = struct{}{}
+
+		go func(address string, podOwner string, deployments []string) {
```

Comment:
this is way better, thanks for doing this. i'm realizing we can take this a step further, and make it even easier to read, what do you think about:

```go
proxy := newSimulatedProxy(randomPodOwner, deployments, sleep)
server := &http.Server{
	Addr:    addr,
	Handler: promhttp.HandlerFor(proxy.registerer, promhttp.HandlerOpts{}),
}
log.Infof("serving scrapable metrics on %s", addr)
go server.ListenAndServe()
go proxy.generateProxyTraffic()
```

Reply:
I like that! Its much easier to follow. I will make the change.